apollo-core/src/test/java/com/ctrip/framework/foundation/internals/UtilsTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.foundation.internals;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class UtilsTest {
+    @Nested
+    class IsBlank {
+        @Test
+        void trueGivenNull() {
+            assertTrue(Utils.isBlank(null));
+        }
+
+        @Test
+        void trueGivenEmpty() {
+            assertTrue(Utils.isBlank(""));
+        }
+
+        @Test
+        void trueGivenWhitespace() {
+            assertTrue(Utils.isBlank("   "));
+        }
+
+        @Test
+        void falseGivenLoremIpsum() {
+            String value = "Lorem Ipsum";
+            assertFalse(Utils.isBlank("Lorem Ipsum"));
+        }
+
+        @Test
+        void falseGivenWhitepsacePadded() {
+            assertFalse(Utils.isBlank("   Lorem Ipsum   "));
+        }
+    }
+
+    @Nested
+    class IsOsWindows {
+        private String actualOsName = System.getProperty("os.name");
+
+        @AfterEach
+        void tearDown() {
+            System.setProperty("os.name", actualOsName);
+        }
+
+        @Test
+        void trueGivenWindows10() {
+            System.setProperty("os.name", "Windows 10");
+            assertTrue(Utils.isOSWindows());
+        }
+
+        @Test
+        void falseGivenMacOsX() {
+            System.setProperty("os.name", "Mac OS X");
+            assertFalse(Utils.isOSWindows());
+        }
+
+        @Test
+        void falseGivenBlank() {
+            System.setProperty("os.name", "");
+            assertFalse(Utils.isOSWindows());
+        }
+
+        // Explicitly calling out case sensitivity; author @nobodyiam should confirm whether this is intentional
+        @Test
+        void falseGivenAllUppercaseWindows() {
+            System.setProperty("os.name", "WINDOWS 10");
+            assertFalse(Utils.isOSWindows());
```

Comment:
I'm not quite sure about thiis, but is the os.name actually in UPPER case for windows 10?

Reply:
It's not - the code actually behaves correctly so these are more like characterization tests. Just emphasizing the case sensitivity since MS could decide to do something wacky in the future 🤷‍♀️